Refine logging infra

mehrwegflasche opened the pull request:
Changes to add exception logging and clean up the controllers.

- Remove repeated try catch with a filter 
- Ensure filter logs to desired sink
- Mark each error with a CorrelationID to aid in future error analysis.

Commit:
Add LoggerInfo and Logger

Commit:
Add Log info , usage loggers ,attributes.

Commit:
Merge branch 'nickpavini:master' into refine-logging-infra

Commit:
Add custom logging to each controller

Commit:
Remove unused Filters

Commit:
remove unused code and add comments

nickpavini commented on SafeAccountsAPI/Program.cs:
```diff
@@ -1,15 +1,19 @@
 ﻿using System;
+using System.IO;
```
I dont think we need either of the new using statements, because no code was added.. we could just get rid of the edits to this page

nickpavini commented on SafeAccountsAPI/SafeAccountsAPI.csproj:
```diff
@@ -18,6 +18,7 @@
     <PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="5.0.0" />
     <PackageReference Include="Serilog.AspNetCore" Version="4.1.0" />
     <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
+    <PackageReference Include="System.Configuration.ConfigurationManager" Version="5.0.0" />
```
since we dont need the extra using statments I dont think we need this either

nickpavini commented on SafeAccountsAPI/Startup.cs:
```diff
@@ -4,6 +4,7 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authentication.JwtBearer;
 using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Diagnostics;
```
I think we also dont need this using statement, or the 3 below

nickpavini commented on SafeAccountsAPI/Controllers/UsersController.cs:
```diff
@@ -192,769 +187,631 @@ private bool ValidatePassword(string input, byte[] storedPassword)
 
         // get a specific accounts info
         [HttpGet("{id:int}/accounts/{account_id:int}")]
+        [ApiExceptionFilter("Error deleting account.")]
```
lol misclick copy pasted the wrong exception message

nickpavini committed:
Update UsersController.cs

Fix an error message, remove a few unneeded summaries, and put exceptionfilters under http attributes

nickpavini committed:
Update Program.cs

remove unneded using statements

nickpavini committed:
Update SafeAccountsAPI.csproj

remove unneeded package

nickpavini committed:
Update Logger.cs

remove unneeded using statements

nickpavini committed:
Update Startup.cs

remove unneeded using statements

nickpavini committed:
Update Startup.cs

nickpavini committed:
Update LoggingInfo.cs
